Utilities/types.h

```diff
@@ -30,7 +30,7 @@
 
 #ifdef _MSC_VER
 
-#define ASSUME(...) __assume(__VA_ARGS__) // MSVC __assume ignores side-effects
+#define ASSUME(...) do { if (!(__VA_ARGS__)) __assume(0); } while(0)  // MSVC __assume ignores side-effects
```

Comment:
Your change renders that comment invalid, doesn't it?

Reply:
The comment is still needed otherwise someone might put the condition back to __assume which is exactly what we are trying to avoid.